exercises/allergies/example.cpp

```diff
@@ -0,0 +1,46 @@
+#include "allergies.h"
+
+#include <string>
+#include <map>
+#include <vector>
+#include <algorithm>
+
+namespace allergies
+{
+
+allergy_test::allergy_test(const int test_result)
+    : result(test_result)
+    , allergens({
+    	{"eggs", 1},
+	{"peanuts", 2},
+	{"shellfish", 4},
+	{"strawberries", 8},
+	{"tomatoes", 16},
+	{"chocolate", 32},
+    	{"pollen", 64},
+	{"cats", 128}
+   })
+{} 
+
+bool allergy_test::is_allergic_to(std::string const& allergen) const
+{
+    int allergen_value = allergens.at(allergen);
+    return (result & allergen_value) == allergen_value;
+}
+
+std::vector<std::string> allergy_test::get_allergies() const
+{
+    std::vector<std::string> allergy_list({"eggs", "peanuts", "shellfish", "strawberries",
+		    			   "tomatoes", "chocolate", "pollen", "cats"});
```

Comment:
Instead of listing all these strings twice (here and in the map), do you think there's a way you can only list them once? Maybe by reading them from the map object here? Then, if you ever needed to update the names or what they mapped to you'd only have to change it in one place.